code/search/CwpSearchEngine.php

```diff
@@ -42,7 +42,12 @@ protected function getSearchQuery($keywords, $classes) {
 		$query->classes = $classes;
 		$query->search($keywords);
 		$query->exclude('SiteTree_ShowInSearch', 0);
-        $query->exclude('File_ShowInSearch', 0);
+
+		// Add File_ShowInSearch if the File class is in the query
+        $classes = array_column($classes, 'class');
+        if (in_array('File', $classes)) {
+            $query->exclude('File_ShowInSearch', 0);
+        }
```

Comment:
I think this is a good patch for now. We could probably loop the classes and check whether they have a ShowInSearch field, then exclude it if it does. At the moment the same problem would apply to anyone not indexing SiteTree (unlikely) too, and the problem of having other DataObjects with ShowInSearch = false set, index and in the query would be the same as File.

We can log this separately (I'll do this).